src/geometry.rs

```diff
@@ -84,7 +84,7 @@ pub enum Value {
 
 impl fmt::Display for Value {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        write!(f, "{:?}", self)
+        write!(f, "{}", Geometry::new(self.clone()))
```

Comment:
thanks for fixing this!

this works as-written, but the `clone()` means we're going to duplicate whatever is currently in memory only to be thrown away immediately after we're done serializing. we do need ownership of the `Value` to create a `Geometry` (which is what you're doing here), but we don't need to ownership to do the underlying serialization operation (we just need to read the contents of `&Value`). we can avoid the `clone()` by copying the code we use for serializing a `Geometry`:

https://github.com/georust/geojson/blob/6a65aad6ff0a89b4a1949b92d2dcb213be85a97d/src/geojson.rs#L328-L350

does this make sense? if you're short on time we can merge-as is and revisit later if that's preferred!

Reply:
Thanks for clarifying! I initially tried that, but I ran into an issue where after implementing it this way:

```rust
impl fmt::Display for Value {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        ::serde_json::to_string(self)
            .map_err(|_| fmt::Error)
            .and_then(|s| f.write_str(&s))
    }
}
```

it only outputs the coordinate array without the geometry type `[[0.0,0.1],[0.1,0.2],[0.2,0.3]]` instead of `{"coordinates":[[0.0,0.1],[0.1,0.2],[0.2,0.3]],"type":"LineString"}"`. Is that alright here, or is there a way to use the geometry serialization without `clone()`?

Reply:
Visualizing the state of things prior to this PR:

|                   | std::fmt::ToString Implementation | serde::Serialize Implementation       |
|-------------------|-----------------------------------|---------------------------------------|
| **GeoJson**           | Delegates to serde::Serialize     | Serializes a GeoJSON object           |
| **Geometry**          | Delegates to serde::Serialize     | Serializes a GeoJSON object           |
| **Feature**           | Delegates to serde::Serialize     | Serializes a GeoJSON object           |
| **FeatureCollection** | Delegates to serde::Serialize     | Serializes a GeoJSON object           |
| **Value**      | Delegates to std::fmt::Debug      | Serializes a GeoJSON coordinate array |

I think this is right? I know this doesn't answer your question, but wanted to share before I dig through the code

Reply:
So yeah you're right. The idea I previously suggested does the wrong thing because it returns a coordinate array and not the full GeoJSON object. A few approaches:

1. Rewrite the `serde::Serialize` implementation for `Value` to also serialize a GeoJSON object, then the `ToString` implementation should do the right thing since it will delegate to `Serialize`
1. Keep the `serde::Serialize` implementation for `Value` the same, but find some other way for the `ToString` implementation to return a GeoJSON string without cloning itself
1. Keep your code as-is with the cloning

I'm fine with any of the strategies above!

I haven't thought for too long about option 1 so not sure how difficult, but worth mentioning the `Serialize` implementation for `Geometry` (and maybe others) [relies on the `Serialize` implementation for `Value` serializing a coordinate array](https://github.com/georust/geojson/blob/c6830fd128eedc68530d439ea634814c635a4d65/src/geometry.rs#L212-L218).

Option 2 _might_ be easier. Here's how I'd imagine doing it:

1. Add a new implementation: `impl<'a> From<&'a Value> for JsonObject`. The implementation should build up a `JsonObject` representing a value GeoJSON object for the `Value`. You can copy/paste most of the existing [`impl<'a> From<&'a Geometry> for JsonObject`](https://github.com/georust/geojson/blob/c6830fd128eedc68530d439ea634814c635a4d65/src/geometry.rs#L193-L226).
1. Update the `std::fmt::ToString` implementation to do `JsonValue::from(self).to_string()`
1. _Extra credit_: Update the `impl<'a> From<&'a Geometry> for JsonObject` to reuse our new `impl<'a> From<&'a Value> for JsonObject` implementation. In particular, the `Geometry` implementation can call the `JsonValue::from(self.value)` (our new implementation), and then proceed to add all the `Geometry` specific fields to the `JsonObject` like `bbox`.

----

As an aside from your pull request, after revisiting our serialization code, it's unfortunate we have to build up an intermediary `JsonValue` for each of our `Serialize` implementations. Probably hurting our performance a bit. Something we can think about in the future.

Reply:
Option 2 makes sense to me, thanks for the detailed explanation of how that could work. I'll take a stab at that and update on progress here

Reply:
This should be implemented in the latest commit, thanks again for the suggestions! I ended up updating `impl<'a> From<&'a Geometry> for JsonObject` as well. Would it make sense to do something similar with `impl TryFrom<JsonObject> for Geometry` using an implementation on `Value`?

Reply:
> Would it make sense to do something similar with `impl TryFrom<JsonObject> for Geometry` using an implementation on `Value`?

if you've got the time and think it'll be an improvement, then i'd say go for it! ✨

Reply:
Sounds good! I looked into this, and it seems like the most straightforward way of implementing `TryFrom<JsonObject> for Value` without copying the data is to add a `utils::get_value` function based on some of the other utility functions that's used by `Value::try_from` and `Geometry::try_from`. I'm thinking this would get pulled out:

```rust
pub fn get_value(object: &mut JsonObject) -> Result<Value, Error> {
    let res = &*expect_type(object)?;
    match res {
        "Point" => Ok(Value::Point(get_coords_one_pos(object)?)),
        "MultiPoint" => Ok(Value::MultiPoint(get_coords_1d_pos(object)?)),
        "LineString" => Ok(Value::LineString(get_coords_1d_pos(object)?)),
        "MultiLineString" => Ok(Value::MultiLineString(get_coords_2d_pos(object)?)),
        "Polygon" => Ok(Value::Polygon(get_coords_2d_pos(object)?)),
        "MultiPolygon" => Ok(Value::MultiPolygon(get_coords_3d_pos(object)?)),
        "GeometryCollection" => Ok(Value::GeometryCollection(get_geometries(object)?)),
        _ => Err(Error::GeometryUnknownType(res.to_string())),
    }
}
```
And then the implementation for `Geometry` would become:

```rust
impl TryFrom<JsonObject> for Geometry {
    type Error = Error;

    fn try_from(mut object: JsonObject) -> Result<Self, Self::Error> {
        let bbox = util::get_bbox(&mut object)?;
        let value = util::get_value(&mut object)?;
        let foreign_members = util::get_foreign_members(object)?;
        Ok(Geometry {
            bbox,
            value,
            foreign_members,
        })
    }
}
```

The `Value` implementation could just call the function directly. Does that sound right? And are there any issues with splitting out `utils::get_value`?

Reply:
looks great to me! i can't think of any issues with that approach

Reply:
Sounds good! That should be added in the latest commit. I also added `from_json_object` and `from_json_value` based on the current setup in `Geometry`, but I can remove those if they aren't needed